### DISCUSSION TITLE
Blazor Server VB.net template

### DISCUSSION THREAD

**Discussion by MarcusMM2** [dotnet/templating#3313] [Template authoring]
Are anyone planning a template for Blazor Server with VB.net (vbproj instead of csproj)?

When I see it right there are only few "things" to collect, maybe I will do it when I know better how to do.

**Comment by KathleenDollard** [comment]
Since VB does not support the Razor engine, we have no plans to create a template. I am checking with the Blazor folks about whether there is a way to use Blazor server without Razor, but I do not think there is.

**Reply by MarcusMM2** [reply]
Hello Kathleen,
thanks for then answer, but to clarify: I am not speaking about a VB-template for the Client-side (WebAssembly) with Razor and @code Parts in C#, I am speaking about the Server part. I think it is not a big deal to write the Controllers in VB (inherits from Microsoft.AspNetCore.Mvc.ControllerBase).

So the question can I write the controllers in VB? For this I need a VB-project (*.vbproj).

**Comment by vlada-shubina** [comment]
I've created https://github.com/dotnet/templating/issues/3351 to create the template. This might be a good opportunity for those willing to contribute and to learn how to create the templates.
